gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,357 @@
+        return self._summary
+
+    def _get_plot_types_for_dim(self, dim: int) -> List[PlotType]:
+        if dim == 0:
+            return [self.PlotType.BAR]
+        if dim == 1:
+            return [self.PlotType.BOX, self.PlotType.HISTOGRAM]
+        return []
+
+    def _get_distribution_histogram(self, data: np.ndarray) -> Dict[str, Union[float, int]]:
+        """Returns the histogram of data as a dictionary.
+
+        If the data is float, the keys of the dictionary are interval buckets, and if the data is int, the keys are also int.
+
+        Args:
+            data: 1D array of all values of the metric
+
+        Returns:
+            Histogram of data as a dict from bucket to count.
+        """
+        if data.size == 0:
+            print("Requested histogram for empty data metric, returning None.")
+            return None
+        if isinstance(data.tolist()[0], int):
+            # One bin for each integer
+            bins = int(np.max(data) - np.min(data) + 1)
+            discrete = True
+        else:
+            bins = 10
+            discrete = False
+        count, bins = np.histogram(data, bins=bins)
+        count = count.tolist()
+        bins = bins.tolist()
+        bins_lower = bins[:-1]
+        bins_upper = bins[1:]
+
+        histogram = {}
+        for i in range(len(count)):
+            if discrete:
+                key = str(int(bins_lower[i]))
+            else:
+                key = "%.2f-%.2f" % (bins_lower[i], bins_upper[i])
+            histogram[key] = count[i]
+        return histogram
+
+    def _create_summary(self, data: np.ndarray) -> Dict[str, Any]:
+        """Creates a summary of the given data.
+
+        This is useful for analysis as data can be very large. The summary is a dict contains the following fields:
+            - Min, max, median of data
+            - Mean and std dev of data
+            - Either quartiles or histogram of the data depending on plot_type of this metric.
+
+        Args:
+            data: 1D array of all values of the metric
+
+        Returns:
+            summary as a dict that can be serialized to JSON for storage.
+        """
+        if data.ndim != 1:
+            raise ValueError("Metric must be a 1D distribution to get summary.")
+        summary = {
+            "min": np.min(data).tolist(),
+            "max": np.max(data).tolist(),
+            "median": np.nanmedian(data).tolist(),
+            "mean": np.nanmean(data).tolist(),
+            "stddev": np.nanstd(data).tolist(),
+        }
+        if self._plot_type == self.PlotType.BOX:
+            summary.update({"quartiles": self._get_distribution_quartiles(data)})
+        elif self._plot_type == self.PlotType.HISTOGRAM:
+            summary.update({"histogram": self._get_distribution_histogram(data)})
+        return summary
+
+    def _get_distribution_quartiles(self, data: np.ndarray) -> Dict[int, float]:
+        """Computes quartiles for the provided 1D data distribution.
```

Comment:
it would be helpful to explain that quartiles are needed for the box plot here, and this is why we compute and save the quartiles.

Reply:
I dont think functions should describe how they are being used. That is to be done at the function call site.

Reply:
i agree it's not strictly required. But I think it would help with readability to just add a 1-liner about why these might be useful.